pkg/providers/snow/snow_test.go

```diff
@@ -0,0 +1,200 @@
+package snow
+
+import (
+	"context"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+
+	"github.com/aws/eks-anywhere/internal/test"
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/providers/docker/mocks"
+	"github.com/aws/eks-anywhere/pkg/types"
+	releasev1alpha1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+const (
+	expectedSnowProviderName  = "snow"
+	testDataDir               = "testdata"
+	testClusterConfigFilename = "cluster_main_1_21.yaml"
+	credsFileEnvVar           = "EKSA_SNOW_DEVICES_CREDENTIALS_FILE"
+	credsFilePath             = "testdata/credentials"
+	certsFileEnvVar           = "EKSA_SNOW_DEVICES_CA_BUNDLES_FILE"
+	certsFilePath             = "testdata/certificates"
+)
+
+type snowTest struct {
+	*WithT
+	ctx         context.Context
+	kubectl     *mocks.MockProviderKubectlClient
+	provider    *snowProvider
+	cluster     *types.Cluster
+	clusterSpec *cluster.Spec
+}
+
+func newSnowTest(t *testing.T) snowTest {
+	ctrl := gomock.NewController(t)
+	kubectl := mocks.NewMockProviderKubectlClient(ctrl)
+	o := givenFullAPIObjects(t, testClusterConfigFilename)
+	cluster := &types.Cluster{
+		Name: "cluster",
+	}
+	provider := newProvider(t, o.datacenterConfig, o.machineConfigs, o.cluster, kubectl)
+	return snowTest{
+		WithT:       NewWithT(t),
+		ctx:         context.Background(),
+		kubectl:     kubectl,
+		provider:    provider,
+		cluster:     cluster,
+		clusterSpec: o.clusterSpec,
+	}
+}
+
+type apiObjects struct {
+	cluster          *v1alpha1.Cluster
+	clusterSpec      *cluster.Spec
+	datacenterConfig *v1alpha1.SnowDatacenterConfig
+	machineConfigs   map[string]*v1alpha1.SnowMachineConfig
+}
+
+func givenFullAPIObjects(t *testing.T, fileName string) apiObjects {
+	clusterSpec := test.NewFullClusterSpec(t, path.Join(testDataDir, fileName))
```

Comment:
Can we build this without using a yaml file? Just defining structs in code
I know that vsphere does it this way but I think structs will be more flexible